### PR TITLE
chore: remove deprected methods plus unused throws

### PR DESCRIPTION
**Pull request by athakor**

### CODE REVIEW

**Review by rahulKQL — CHANGES_REQUESTED** [review]
Please divide this PR into two or more PRs.
 1. Which removes throws clause
 2. Removes deprecated methods
 3. If any other possible combination

**Comment by athakor** [comment]
> Please divide this PR into two or more PRs.
> 
> 1. Which removes throws clause
> 2. Removes deprecated methods
> 3. If any other possible combination

1.[PR](https://github.com/athakor/java-datastore/pull/5) for to removes throws clause
2.[PR](https://github.com/athakor/java-datastore/pull/4) for to removes deprecated methods

File: google-cloud-datastore/src/test/java/com/google/cloud/datastore/ITDatastoreTest.java [hunk 1]
```diff
@@ -1108,30 +1104,36 @@ public void testRetryableException() throws Exception {
...
-    thrown.expectMessage("denied");
+    Datastore datastore = null;
+    try {
+      datastore = rpcMockOptions.getService();
+    } catch (DatastoreException e) {
+      assertEquals("denied", e.getMessage());
+    }
     datastore.get(KEY1);
     EasyMock.verify(rpcFactoryMock, rpcMock);
   }
```
Comment (dmitry-fa) [inline google-cloud-datastore/src/test/java/com/google/cloud/datastore/ITDatastoreTest.java:1123]: not sure if `expected = DatastoreException.class` is needed here and in the test below.
Does `EasyMock.verify(rpcFactoryMock, rpcMock)` throws `DatastoreException`?

File: google-cloud-datastore/src/test/java/com/google/cloud/datastore/LatLngTest.java [hunk 2]
```diff
@@ -47,35 +43,43 @@ public void testEquals() {
...
-    new LatLng(90, 0);
-    thrown.expect(IllegalArgumentException.class);
-    thrown.expectMessage(INVALID_LAT_MESSAGE);
+    try {
+      new LatLng(90, 0);
+    } catch (IllegalArgumentException e) {
+      assertEquals(INVALID_LAT_MESSAGE, e.getMessage());
+    }
     new LatLng(91, 0);
   }
```
Comment (dmitry-fa) [inline google-cloud-datastore/src/test/java/com/google/cloud/datastore/LatLngTest.java:55]: In LatLngTest for IllegalArgumentException the negative cases is checked twice. That's not needed.
The test will look better if you remove the last constructor call

```
@Test // no expected exception here
public void testUpperLatRange() {
 try {
      new LatLng(90, 0);
    } catch (IllegalArgumentException e) {
      assertEquals(INVALID_LAT_MESSAGE, e.getMessage());
    }
  }
}
```

**Pull request closed without merging**